### PR TITLE
Add backendRoles to userrolebinding

### PR DESCRIPTION
**Pull request by AleksZimin**
Fixes https://github.com/Opster/opensearch-k8s-operator/issues/294

### CODE REVIEW

File: charts/opensearch-operator/templates/OpensearchUserRoleBinding-crd.yaml [hunk 1]
```diff
@@ -58,7 +62,6 @@ spec:
             required:
             - opensearchCluster
             - roles
-            - users
```
Comment (idanl21) [inline charts/opensearch-operator/templates/OpensearchUserRoleBinding-crd.yaml:61]: Didn't understand why to delete the users?

File: opensearch-operator/config/crd/bases/opensearch.opster.io_opensearchuserrolebindings.yaml [hunk 2]
```diff
@@ -60,7 +64,6 @@ spec:
             required:
             - opensearchCluster
             - roles
-            - users
```
Comment (idanl21) [inline opensearch-operator/config/crd/bases/opensearch.opster.io_opensearchuserrolebindings.yaml:63]: same here

File: opensearch-operator/pkg/reconcilers/userrolebinding.go [hunk 3]
```diff
@@ -182,19 +184,10 @@ func (r *UserRoleBindingReconciler) Reconcile() (retResult ctrl.Result, retErr e
...
-					reason = "failed to update existing role mapping"
-					r.logger.Error(retErr, reason)
-					r.recorder.Event(r.instance, "Warning", opensearchAPIError, reason)
-					return
-				}
-			}
-			// Then add new users
-			retErr = r.reconcileExistingMapping(role)
+			// Replace existing mapping with new one
+			removedUsers := helpers.DiffSlice(r.instance.Status.ProvisionedUsers, r.instance.Spec.Users)
```
Comment (idanl21) [inline opensearch-operator/pkg/reconcilers/userrolebinding.go:188]: good improvement

File: charts/opensearch-operator/templates/OpensearchUserRoleBinding-crd.yaml (hunk 1 above)
Comment (AleksZimin) [reply charts/opensearch-operator/templates/OpensearchUserRoleBinding-crd.yaml:61]: Hello! We can add only users or only backendRoles to mapping. So I delete users from requered fields. Maybe we can implement choice between backendRoles and users here... But I don't know how to do this.

I did some manual tests without users and backendRoles fields in OpensearchUserRoleBinding manifest like this:
```yaml
apiVersion: opensearch.opster.io/v1
kind: OpensearchUserRoleBinding
metadata:
  name: empty-rolebinding
spec:
  opensearchCluster:
    name: opensearch-cluster
  roles:
  - test-role

```
and didn't get any errors.

File: opensearch-operator/pkg/reconcilers/userrolebinding.go (hunk 3 above)
Comment (AleksZimin) [reply opensearch-operator/pkg/reconcilers/userrolebinding.go:188]: Thanks :)

File: charts/opensearch-operator/templates/OpensearchUserRoleBinding-crd.yaml (hunk 1 above)
Comment (swoehrl-mw) [reply charts/opensearch-operator/templates/OpensearchUserRoleBinding-crd.yaml:61]: IMO a combination of users and backendRoles is possible. And if you tested and leaving out both users and backendRoles doesn't produce errors then it's ok for me.

**Comment by swoehrl-mw** [comment]
@idanl21 @prudhvigodithi Are you OK to merge this one?

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Hey @swoehrl-mw i'm good with this. Thanks for the PR @AleksZimin

**Pull request merged**